realm/realm-library/src/androidTest/java/io/realm/RealmListTests.java

```diff
@@ -1053,17 +1053,24 @@ public void onChange(RealmList<Dog> collection, OrderedCollectionChangeSet chang
                 fail();
             }
         });
-        realm.beginTransaction();
-        collection.get(0).setAge(42);
-        realm.commitTransaction();
 
         collection.removeAllChangeListeners();
 
+        // This one is added after removal, so it should be triggered.
+        collection.addChangeListener(new RealmChangeListener<RealmList<Dog>>() {
```

Comment:
because of i don't want to rely on the behavior "listeners triggered when commit" in this test. By adding another listener after all listeners removed, it can ensure that `removeAllChangeListeners` works, and the listener added after will be triggered. Which means the same test should work even when "listeners are triggered in the next event loop."